torch/nn/functional.py

```diff
@@ -1900,7 +1918,14 @@ def mse_loss(input, target, size_average=None, reduce=None, reduction='mean'):
     """
     if size_average is not None or reduce is not None:
         reduction = _Reduction.legacy_get_string(size_average, reduce)
-    return _pointwise_loss(lambda a, b: (a - b) ** 2, torch._C._nn.mse_loss, input, target, reduction)
+    if target.requires_grad:
+        ret = (input - target) ** 2
```

Comment:
@li-roy: do you remember why we do this? I find it weird that we can't put all of this into aten

Reply:
It's because we have optimized kernels in THNN which only handle a subset of the cases. If we wanted to migrate it to ATen, we would either have a perf regression, or have to write a lot of kernels to handle those cases.

Reply:
We could also leverage THNN kernels from ATen, but I'm not sure we would gain much from it (apart from c++ parity)

Reply:
That's unfortunate, thanks for the context @fmassa